model-integration/src/main/java/ai/vespa/embedding/VoyageAIEmbedder.java

```diff
@@ -0,0 +1,400 @@
+ * @author VoyageAI team
+ */
+@Beta
+public class VoyageAIEmbedder extends AbstractComponent implements Embedder {
+
+    private static final Logger log = Logger.getLogger(VoyageAIEmbedder.class.getName());
+    private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    // Configuration
+    private final VoyageAiEmbedderConfig config;
+    private final Embedder.Runtime runtime;
+    private final Secret apiKey;
+    private final OkHttpClient httpClient;
+
+    // Cache for embeddings
+    private final Map<CacheKey, Tensor> embeddingCache;
+
+    @Inject
+    public VoyageAIEmbedder(VoyageAiEmbedderConfig config, Embedder.Runtime runtime, Secrets secretStore) {
+        this.config = config;
+        this.runtime = runtime;
+        this.apiKey = getApiKey(config, secretStore);
+        this.httpClient = createHttpClient(config);
+        this.embeddingCache = config.cacheSize() > 0
+                ? new LRUCache<>(config.cacheSize())
+                : new ConcurrentHashMap<>();
+
+        log.info("VoyageAI embedder initialized with model: " + config.model());
+    }
+
+    /**
+     * Retrieve API key from Vespa's secret store.
+     */
+    private Secret getApiKey(VoyageAiEmbedderConfig config, Secrets secretStore) {
+        String secretName = config.apiKeySecretRef();
+
+        if (secretName == null || secretName.isEmpty()) {
+            throw new IllegalArgumentException(
+                "api-key-secret-ref must be configured for VoyageAI embedder. " +
+                "Please set it in services.xml and ensure the secret is in the secret store."
+            );
+        }
+
+        try {
+            Secret secret = secretStore.get(secretName);
+            if (secret == null) {
+                throw new IllegalArgumentException(
+                    "Secret not found in secret store: " + secretName + ". " +
+                    "Please add it using: vespa secret add " + secretName + " --value YOUR_API_KEY"
+                );
+            }
+            return secret;
+        } catch (UnsupportedOperationException e) {
+            throw new IllegalArgumentException(
+                "Secret store is not configured. Cannot retrieve API key for VoyageAI embedder.", e
+            );
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                "Failed to retrieve API key from secret store. Secret name: " + secretName, e
+            );
+        }
+    }
+
+    /**
+     * Create HTTP client with connection pooling and timeouts.
+     */
+    private OkHttpClient createHttpClient(VoyageAiEmbedderConfig config) {
+        return new OkHttpClient.Builder()
+                .connectTimeout(Duration.ofMillis(config.timeout()))
+                .readTimeout(Duration.ofMillis(config.timeout()))
+                .writeTimeout(Duration.ofMillis(config.timeout()))
```

Comment:
Please specify the `callTimeout` as well for a proper request-response timeout on the client side. 

We want to later include a timeout value in the `Context`, so timeout can be tied to the logical operation (respect overall feed/query time). For now we'll use the timeout configured in the embedder, but later will override the call timeout per request and use the lowest of those two.